thanks for catching that. might need some tweaks to keep cross-compilation working though:

```
$ git fetch upstream pull/369595/head:pr-369595
$ git switch pr-369595
$ nix-build -A pkgsCross.aarch64-multiplatform.hspell
...
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/cc -c -g -O2  -DPACKAGE_NAME=\"\" -DPACKAGE_TARNAME=\"\" -DPACKAGE_VERSION=\"\" -DPACKAGE_STRING=\"\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_INTPTR_T=1 -DHAVE_ZLIB=1 -DDICTIONARY_BASE=\"/nix/store/lfaqm62r958izwxsh7d52nzl529j7glz-hspell-1.4-aarch64-unknown-linux-gnu/share/hspell/hebrew.wgz\" dict_radix.c
In file included from dict_radix.c:34:
gzbuffered.h:30:10: fatal error: zlib.h: No such file or directory
   30 | #include <zlib.h>
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:73: dict_radix.o] Error 1
error: builder for '/nix/store/vydr2z1scvk0vh4hqhp71d4r3sx613p9-hspell-1.4-aarch64-unknown-linux-gnu.drv' failed with exit code 2;
```

--- a/pkgs/development/libraries/hspell/default.nix
+++ b/pkgs/development/libraries/hspell/default.nix
@@ -25,7 +25,7 @@ stdenv.mkDerivation rec {
   patches = [ ./remove-shared-library-checks.patch ];
   postPatch = "patchShebangs .";
   preBuild = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
-    make CC=${buildPackages.stdenv.cc}/bin/cc find_sizes
+    make CC='${buildPackages.stdenv.cc}/bin/cc -I${lib.getDev buildPackages.zlib}/include -L${buildPackages.zlib}/lib' find_sizes
     mv find_sizes find_sizes_build
     make clean
 
@@ -41,7 +41,10 @@ stdenv.mkDerivation rec {
     perl
     zlib
   ];
-  buildInputs = [ perl ];
+  buildInputs = [
+    perl
+    zlib
+  ];
 
   strictDeps = true;
 

